src/pages/Pools/Pair/index.tsx

```diff
@@ -63,12 +63,10 @@ const ButtonRow = styled(RowFixed)`
   `};
 `
 
-export default function Pair({
-  match: {
-    params: { currencyIdA, currencyIdB },
-  },
-}: RouteComponentProps<{ currencyIdA: string; currencyIdB: string }>) {
-  const router = useRouter()
+export default function Pair() {
+  const { navigate, searchParams: search } = useRouter()
+  const { currencyIdA, currencyIdB } = useParams<{ currencyIdA: string; currencyIdB: string }>()
```

Comment:
This type could be moved to dedicated interface as it's used really often

```ts
//...somewhere in the codebase
export interface CurrencySearchParams {
   currencyIdA: string
   currencyIdB: string
}
```

Reply:
I will move it a type in each Component. I feel keeping types close to the component for easy code reading and maintenance

Reply:
Okay, but what about DRY?